cg/cli/workflow/microsalt/base.py

```diff
@@ -10,149 +10,117 @@
     """Link microbial FASTQ files for a SAMPLE_ID"""
-    sample_objs = None
-
-    if order_id and (sample_id is None):
-        # link all samples in a case
-        sample_objs = context.obj["db"].microbial_order(order_id).microbial_samples
-    elif sample_id and (order_id is None):
-        # link sample in all its families
-        sample_objs = [context.obj["db"].microbial_sample(sample_id)]
-    elif sample_id and order_id:
-        # link only one sample in a case
-        sample_objs = [context.obj["db"].microbial_sample(sample_id)]
+
+    api = context.obj["api"]
+    sample_objs = api.get_samples(ticket, sample_id)
 
     if not sample_objs:
-        LOG.error("provide order and/or sample")
+        LOG.error("provide ticket and/or sample")
         context.abort()
 
     for sample_obj in sample_objs:
         LOG.info("%s: link FASTQ files", sample_obj.internal_id)
-        context.obj["api"].link_sample(
+        if dry_run:
+            continue
+        api.link_sample(
             FastqHandler(context.obj),
-            case=sample_obj.microbial_order.internal_id,
+            case=ticket,
             sample=sample_obj.internal_id,
         )
 
 
 @microsalt.command("config-case")
-@click.option("-d", "--dry", is_flag=True, help="print config-case to console")
-@click.option(
-    "-o",
-    "--order",
-    "order_id",
-    help="create config-case all microbial samples for an order",
-)
+@click.option("-d", "--dry-run", is_flag=True, help="print config to console")
+@click.option("-t", "--ticket", help="create config-case all microbial samples for an order")
 @click.argument("sample_id", required=False)
 @click.pass_context
-def config_case(context: click.Context, dry, order_id: str, sample_id: str):
+def config_case(context: click.Context, dry_run, ticket: int, sample_id: str):
     """ Create a config file on case level for microSALT """
-    if order_id and (sample_id is None):
-        microbial_order_obj = context.obj["db"].microbial_order(order_id)
-        if not microbial_order_obj:
-            LOG.error("Order %s not found", order_id)
-            context.abort()
-        sample_objs = microbial_order_obj.microbial_samples
-    elif sample_id and (order_id is None):
-        sample_obj = context.obj["db"].microbial_sample(sample_id)
-        if not sample_obj:
-            LOG.error("Sample %s not found", sample_id)
-            context.abort()
-        sample_objs = [context.obj["db"].microbial_sample(sample_id)]
-    elif sample_id and order_id:
-        microbial_order_obj = context.obj["db"].microbial_order(order_id)
-        if not microbial_order_obj:
-            LOG.error("Samples %s not found in %s ", sample_id, order_id)
-            context.abort()
-        sample_objs = [
-            sample_obj
-            for sample_obj in microbial_order_obj.microbial_samples
-            if sample_obj.internal_id == sample_id
-        ]
-    else:
-        LOG.error("provide order and/or sample")
+
```

Comment:
Really nice job on the code reduction/refactoring in these CLI commands @patrikgrenfeldt !